common/build-style/zig-build.sh

```diff
@@ -0,0 +1,41 @@
+do_build() {
+	local zig_target zig_cpu
+
+	# TODO: This duplication between build-profiles and cross-profiles
+	# is totally unnecessary. It would be nice if there was some way to
+	# avoid it.
+	if [ "$CROSS_BUILD" ]; then
+		zig_target="${XBPS_CROSS_ZIG_TARGET}"
+		zig_cpu="${XBPS_CROSS_ZIG_CPU}"
+	else
+		zig_target="${XBPS_ZIG_TARGET}"
+		zig_cpu="${XBPS_ZIG_CPU}"
+	fi
+
+	# Inform zig of the required libc include paths.
+	cat > xbps_zig_libc.txt <<-EOF
+		include_dir=${XBPS_CROSS_BASE}/usr/include
+		sys_include_dir=${XBPS_CROSS_BASE}/usr/include
+		crt_dir=${XBPS_CROSS_BASE}/usr/lib
+		msvc_lib_dir=
+		kernel32_lib_dir=
+		gcc_dir=
```

Comment:
Would it make sense to query gcc_dir from GCC itself? Would it be worth it?

Reply:
according to the output of `zig libc`, `gcc_dir` is only needed when targeting haiku:
```
# The directory that contains `crtbeginS.o` and `crtendS.o`
# Only needed when targeting Haiku.
gcc_dir=
```